src/Detector/Go/GoDetector.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Common.Extensions;
+using System;
+using System.Linq;
+
+namespace Microsoft.Oryx.Detector.Go
+{
+    /// <summary>
+    /// An implementation of <see cref="IPlatformDetector"/> which detects Go applications.
+    /// </summary>
+    class GoDetector : IGoPlatformDetector
+    {
+        private readonly ILogger<GoDetector> _logger;
+        private readonly DetectorOptions _options;
+
+        /// <summary>
+        /// Creates an instance of <see cref="GoDetector"/>.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger{GoDetector}"/>.</param>
+        /// <param name="options">The <see cref="DetectorOptions"/>.</param>
+        public GoDetector(ILogger<GoDetector> logger, IOptions<DetectorOptions> options)
+        {
+            _logger = logger;
+            _options = options.Value;
+        }
+
+        public PlatformDetectorResult Detect(DetectorContext context)
+        {
+            bool isGoApp = false;
+            string appDirectory = string.Empty;
+            var sourceRepo = context.SourceRepo;
+            bool goDotModExists = false;
+            // check if go.mod exists
+            if (sourceRepo.FileExists(GoConstants.GoDotModFileName))
+            {
+                goDotModExists = true;
+                isGoApp = true;
+                _logger.LogInformation($"Found {GoConstants.GoDotModFileName} at the root of the repo. ");
+            }
+            else
+            {
+                _logger.LogDebug(
+                    $"Could not find {GoConstants.GoDotModFileName} in repo");
+                return null;
+            }
+
+            // TODO: check if go.sum or /cmd/repo_name/*.go exists
+
+            var version = GetVersion(context);
+            return new GoPlatformDetectorResult
+            {
+                Platform = GoConstants.PlatformName,
+                PlatformVersion = version,
+                AppDirectory = appDirectory,
+            };
+        }
+
+        private string GetVersion(DetectorContext context)
+        {
+            var versionFromGoDotMod = GetVersionFromGoDotMod(context);
+            if (versionFromGoDotMod != null)
+            {
+                return versionFromGoDotMod;
+            }
+            return null;
```

Comment:
You should be able to just do the following here:

```cs
return versionFromGoDotMod;
```

Again, this will change if you introduce different files to check 😁

Reply:
https://github.com/microsoft/Oryx/pull/1071#discussion_r681340742